### ISSUE TITLE
"Pause" does not work anymore

### ISSUE THREAD

**Issue by d01010101** [apache/netbeans#5356]
### Apache NetBeans version

Apache NetBeans 16

### What happened

Did "Pause" actually ever go to the paused statement, as in case of other IDEs?  I may be wrong, but I seem to recall that it was the case. Now it works just never, no matter which project. Pressing "Pause" stops the execution, enables the "Continue" button and the latter is the only visible effect. I still have no idea which code line or even thread has been paused.

### How to reproduce

Just run the debugger and press "Pause". It may be an Ant single-thread project, a Maven project with multiple user threads, does not matter.

### Did this work correctly in an earlier version?

No / Don't know

### Operating System

Ubuntu 20.04

### JDK

OpenJDK 64-Bit Server VM (build 14.0.1+7, mixed mode, sharing)

### Apache NetBeans packaging

Apache NetBeans provided installer

### Are you willing to submit a pull request?

No

### Code of Conduct

Yes

**Comment by neilcsmith-net** [comment]
It sounds like you're expecting `Pause` to function like hitting a breakpoint?  You can use the debugger tab to see all paused threads, where they're paused, and jump to the required code lines in each.

**Comment by d01010101** [comment]
Why it does not happen automatically if there is only one running thread within `main` and threads created by `main`?

**Comment by d01010101** [comment]
Or alternately, if there is only one thread running the code of currently open projects?

**Comment by neilcsmith-net** [comment]
Personally, I don't see the benefit in doing this, and on only the simplest projects could it do anything useful.  I'm also not sure it's a bug.  Moving to feature request, but leaving open in case anyone thinks it's something worth pursuing.

**Comment by d01010101** [comment]
Can you justify your conclusion that the projects, which do not have most of the time multiple threads running at the same time the code which is currently open, are "only the simplest projects"?
This is my second bug report. In the case of the first one, I made a list of several arguments. Your argument was "I like it as it is now". I won't even attempt to try to convince you the second time.

**Comment by neilcsmith-net** [comment]
Yes, I can.  Still, it's not about convincing me, it's about either putting in a pull request or convincing someone else that this is a feature worth looking.  And on that basis, reopening.

**Comment by lkishalmi** [comment]
I do see that Pause , pauses all the threads, and you can examine the execution stack on each and every thread. Usually there are a handful of them. I would not want to inspect each and every one where it has stopped. It is also a pretty rare use case to do in a debugger (when hunting on a bad concurrency issue). So if anyone would like to contribute such a functionality, please consider that as optional.

**Comment by d01010101** [comment]
Did you even read what this issue was about? It was specifically about the case with a single thread running. Actually running, as in using CPU by going through the user code. Something deemed by your friend as "only a simplest project". Which would make Netbeans itself even more simple than "only a simplest project" because normally it takes 0% CPU.

**Comment by neilcsmith-net** [comment]
There's always more than a single thread in NetBeans itself, even when the IDE is using minimal CPU.  It's even less clear to me what you think this should be doing now.

**Comment by d01010101** [comment]
I closed this issue seeing how the discussion is going. Ok, maybe they think it is useless, not a problem. Then you reopen it to talk about "any thread" in an answer to a message which said clearly that it is not about "any thread".

In IntelliJ, and most IDEs which I used, pausing navigates directly to a thread. But not to any thread. There is some logic to make it useful. I asked about a feature commonly implemented in other IDEs for user's convenience. See for example here https://www.youtube.com/watch?v=lAWnIP1S6UA&t=54s
